src/modules/esl-carousel/plugin/touch/esl-carousel.touch.mixin.ts

```diff
@@ -90,11 +115,32 @@ export class ESLCarouselTouchMixin extends ESLCarouselPlugin {
     if (this.$$attr('dragging', false) !== null) {
       event.preventDefault();
       const point = getTouchPoint(event);
-      const offset = point.x - this.startPoint.x;
+      const offset = this.$host.config.vertical ? point.y - this.startPoint.y : point.x - this.startPoint.x;
       // ignore single click
       offset !== 0 && this.$host.renderer.commit(offset);
     }
   }
+
+  /** Handles `swipe` event */
+  @listen({
+    event: ESLSwipeGestureEvent.type,
+    target: ESLSwipeGestureTarget.for,
+    condition: (that: ESLCarouselTouchMixin)=> that.typeRule.value === ESLCarouselTouchMixin.SWIPE_TYPE
+  })
+  protected _onSwipe(e: ESLSwipeGestureEvent): void {
+    if (!this.$host || this.$host.animating) return;
+    if (this.$host.config.vertical !== e.isVertical) return;
+    const direction = e.direction === 'left' || e.direction === 'up' ? 'next' : 'prev';
+    this.$host?.goTo(`${this.swipeType}:${direction}`);
```

Comment:
```suggestion
    const {$host} = this;
    if (!$host || $host.animating) return;
    if ($host.config.vertical !== e.isVertical) return;
    const direction = e.direction === 'left' || e.direction === 'up' ? 'next' : 'prev';
    $host?.goTo(`${this.swipeType}:${direction}`);
```

Reply:
`$host` is not computed value, so that is totally cosmetic thing